src/DocBlock/TypeExpression.php

```diff
@@ -63,13 +63,19 @@ final class TypeExpression
                     \h*\}
                 )
                 |
-                (?<callable> # callable syntax, e.g. `callable(string): bool`
-                    (?<callable_start>(?i)(?:callable|\\\\?Closure)(?-i)\h*\(\h*)
+                (?<callable> # callable syntax, e.g. `callable(string, int...): bool`
+                    (?<callable_start>(?&name)\h*\(\h*)
                     (?<callable_arguments>
-                        (?&types_inner)
+                        (?<callable_argument>
+                            (?<callable_argument_type>(?&types_inner))
+                            (?<callable_argument_is_reference>\h*&|)
```

Comment:
- shouldn't passing by reference require identifier? I don't see [PHPStan example](https://phpstan.org/writing-php-code/phpdoc-types#callables) with `type&` only (contrary to example with variadic without identifier)
- shouldn't passing by reference be disjunctive with variadic param? I was pretty sure it would be an syntax error, but it looks like it "works" (well, [does not end with fatal error, but value is not re-assigned](https://twitter.com/_Codito_/status/1674370675639820288)).

Reply:
In parser we are not required to "not-match any invalid" edge usecase, we are required to "match all valid" usecases.

I coded it based on phpstan parser grammar.

To your bullet points - can you please provided phpstan and psalm online repro links, so we can see how the static analysers really understand it (or emit warnigs).